pkg/common/cns-lib/volume/manager.go

```diff
@@ -496,6 +537,68 @@ func (m *defaultManager) createVolumeWithImprovedIdempotency(ctx context.Context
 	return resp, faultType, err
 }
 
+func (m *defaultManager) waitOnTask(ctx context.Context, taskMoRef vim25types.ManagedObjectReference) (*vim25types.TaskInfo, error) {
+	listViewCtx := context.TODO()
+	log := logger.GetLogger(listViewCtx)
+	if m.taskViewManager == nil {
+		log.Infof("wait on task manager is nil")
+		err := m.initTaskViewManager(listViewCtx)
+		if err != nil {
+			return nil, err
+		}
+	}
+	ch := m.taskViewManager.GetNewResultChannelForCaller()
+	err := m.taskViewManager.AddTaskToListView(listViewCtx, taskMoRef, ch)
+	if err != nil {
```

Comment:
Can you give an example of how this is harmless, if there is an error adding the task to list view, how are we sure that there won't be runaway tasks that may lead to orphan volumes or snaphots?

Reply:
IIUC, create volume and snapshots both are called with improved idempotency in which case there won't be a new CNS task created when they are retried. Is that accurate?